images/router/haproxy/conf/haproxy-config.template

```diff
@@ -655,8 +658,14 @@ backend {{genBackendNamePrefix $cfg.TLSTermination}}:{{$cfgIdx}}
       {{- end }}
   tcp-request content reject if !whitelist
     {{- end }}
-    {{- with $value := clipHAProxyTimeoutValue (firstMatch $timeSpecPattern (index $cfg.Annotations "haproxy.router.openshift.io/timeout")) }}
+    {{- if (index $cfg.Annotations "haproxy.router.openshift.io/timeout-tunnel") }}
+      {{- with $value := clipHAProxyTimeoutValue (firstMatch $timeSpecPattern (index $cfg.Annotations "haproxy.router.openshift.io/timeout-tunnel")) }}
   timeout tunnel  {{$value}}
+      {{- end }}
+    {{- else }}
+      {{- with $value := clipHAProxyTimeoutValue (firstMatch $timeSpecPattern (index $cfg.Annotations "haproxy.router.openshift.io/timeout")) }}
+  timeout tunnel  {{$value}}
+      {{- end }}
     {{- end }}
```

Comment:
I believe this could be simplified slightly by deleting the first `{{ if }}` and raising the nested `{{ with }}`:
```go
    {{- with $value := clipHAProxyTimeoutValue (firstMatch $timeSpecPattern (index $cfg.Annotations "haproxy.router.openshift.io/timeout-tunnel")) }}
  timeout tunnel  {{$value}}
    {{- else }}
      {{- with $value := clipHAProxyTimeoutValue (firstMatch $timeSpecPattern (index $cfg.Annotations "haproxy.router.openshift.io/timeout")) }}
  timeout tunnel  {{$value}}
      {{- end }}
    {{ -end }}
```
(Unfortunately, there is no `{{ else with ... }}` according to <https://golang.org/pkg/text/template/>, which would facilitate further simplification.)

Reply:
Actually, the following would be even simpler:
```go
    {{- with $value := clipHAProxyTimeoutValue (firstMatch $timeSpecPattern (index $cfg.Annotations "haproxy.router.openshift.io/timeout-tunnel") (index $cfg.Annotations "haproxy.router.openshift.io/timeout")) }}
  timeout tunnel  {{$value}}
    {{ -end }}
```

Reply:
Ah, great. Prefer the combined/simpler version. Thanks!